spec/01-unit/002-conf_loader_spec.lua

```diff
@@ -73,16 +69,54 @@ describe("Configuration loader", function()
     assert.True(conf.plugins["hello-world"])
     assert.True(conf.plugins["another-one"])
   end)
-  it("extracts ports and listen ips from proxy_listen/admin_listen", function()
+  it("extracts flags, ports and listen ips from proxy_listen/admin_listen", function()
     local conf = assert(conf_loader())
-    assert.equal("127.0.0.1", conf.admin_ip)
-    assert.equal(8001, conf.admin_port)
-    assert.equal("127.0.0.1", conf.admin_ssl_ip)
-    assert.equal(8444, conf.admin_ssl_port)
-    assert.equal("0.0.0.0", conf.proxy_ip)
-    assert.equal(8000, conf.proxy_port)
-    assert.equal("0.0.0.0", conf.proxy_ssl_ip)
-    assert.equal(8443, conf.proxy_ssl_port)
+    assert.equal("127.0.0.1", conf.admin_listeners[1].ip)
+    assert.equal(8001, conf.admin_listeners[1].port)
+    assert.equal(false, conf.admin_listeners[1].ssl)
+    assert.equal(false, conf.admin_listeners[1].http2)
+    assert.equal("127.0.0.1:8001", conf.admin_listeners[1].listener)
+
+    assert.equal("127.0.0.1", conf.admin_listeners[2].ip)
+    assert.equal(8444, conf.admin_listeners[2].port)
+    assert.equal(true, conf.admin_listeners[2].ssl)
+    assert.equal(false, conf.admin_listeners[2].http2)
+    assert.equal("127.0.0.1:8444 ssl", conf.admin_listeners[2].listener)
+
+    assert.equal("0.0.0.0", conf.proxy_listeners[1].ip)
+    assert.equal(8000, conf.proxy_listeners[1].port)
+    assert.equal(false, conf.proxy_listeners[1].ssl)
+    assert.equal(false, conf.proxy_listeners[1].http2)
+    assert.equal("0.0.0.0:8000", conf.proxy_listeners[1].listener)
+
+    assert.equal("0.0.0.0", conf.proxy_listeners[2].ip)
+    assert.equal(8443, conf.proxy_listeners[2].port)
+    assert.equal(true, conf.proxy_listeners[2].ssl)
+    assert.equal(false, conf.proxy_listeners[2].http2)
+    assert.equal("0.0.0.0:8443 ssl", conf.proxy_listeners[2].listener)
+  end)
+  it("extracts 'off' from proxy_listen/admin_listen", function()
+      local conf
+    conf = assert(conf_loader(nil, {
+      proxy_listen = "off",
+      admin_listen = "off",
+    }))
+    assert.same({}, conf.proxy_listeners)
+    assert.same({}, conf.admin_listeners)
+    -- off with multiple entries
+    conf = assert(conf_loader(nil, {
+      proxy_listen = "off, 0.0.0.0:9000",
+      admin_listen = "off, 127.0.0.1:9001",
+    }))
+    assert.same({}, conf.proxy_listeners)
+    assert.same({}, conf.admin_listeners)
+    -- not off with names containing 'off'
```

Comment:
I don't see the same test for a hostname containing 'ssl' (with a key and no cert defined, so it returns the appropriate configuration error) (for both admin and proxy interfaces). Shall we add it?